### DISCUSSION TITLE
Snipe-it -Docker SSL

### DISCUSSION THREAD

**Discussion by Puma4ka** [snipe/snipe-it#13484] [Show and tell]
can't understand how to connect ssl correctly.

I have *****.cer ******.key files which I have placed in /var/lib/snipeit/ssl
and I don't really understand what to do next!?!?!

At the moment, the page worked for me as http://, but I need https://

this is my docker-compose.yml

![1snipe](https://github.com/snipe/snipe-it/assets/66869511/920cfa4e-ea61-4998-a33c-12b8322be718)

and my emv.docker

![2snipe](https://github.com/snipe/snipe-it/assets/66869511/00a8da03-df85-4b37-8eb7-96d6dd00bb34)

APP_URL= I tried to change https://
but of course without results. what should I do and where should I complete so that I can finally start working on https?

**Comment by nellman-One** [comment]
Hi!
I have the same issue.
Did you get this to work properly?

**Comment by Puma4ka** [comment]
hi! unfortunately no, it's not possible to solve it.

**Reply by nellman-One** [reply]
Hello! I agree. But it has to work somehow... but obviously there is something I am doing very wrong.

**Reply by Puma4ka** [reply]
![Porti](https://github.com/snipe/snipe-it/assets/66869511/9fd44e30-4578-4182-9271-3265c5ac804e)

I really don't understand what the problem is. that you can see picture 443 is attached. I tried to change the env file to https, I removed the 80:80 port from the docker-compose file, leaving only 443, but it still doesn't work. ssl files are located in the place where they should have both key and crt file.

**Comment by xorist** [comment]
@Puma4ka
@nellman-One 

I figured out how to fix this, and I noticed that some other people were having issues terminating SSL at snipe instead of using a reverse proxy (RP). Below are the steps I took to fix this issue.

1. I cloned the repository
2. I edited the docker-compose.yml file to use the dockerfile called "Dockerfile" that comes downloaded with the github repo as of release 6.2.3, instead of the Dockerfile.alpine that is used by default. This is because the "Dockerfile" dockerfile already contains a lot of the setup necessary for supporting SSL, unlike the "Dockerfile.alpine" dockerfile.
3. I mapped a local directory that contained my SSL certificates to the container's directory that is referenced by the "docker/001-default-ssl.conf" Apache site configuration file included with the downloaded repo. I also make sure they had the same names as referenced by the file. The destination directory within the container should be "/var/lib/snipeit/ssl", without quotes, of course. The source should again be the directory on your host that contains your SSL Certificates.
4. I added the port mapping 443:443 in the docker-compose.yml file as well.

After all that, you should be able to bring up the container and have your port 443 serving HTTPS with your certificate. If you've already set up SnipeIT to use a reverse proxy, or are just using it via HTTP, you will have to blow away your image and volumes and start over. Make sure to back up your data before doing this, so you can restore it later. The reason for removing the image is that the image will be re-created using the "Dockerfile" dockerfile, instead of the "Dockerfile.alpine" dockerfile. The reason for removing the volumes is that some things in the volume reference the old port you were using. This will cause some JavaScript and CSS to not load properly until it's changed to the new port. Finally, make sure that in your .env.docker file, you change the APP_URL value to the new "https" version of your snipe instance.
